File: 013 Palindome Check/Palindrome check.js
// O(n) Time and Space
function isPalindrome(string) {
	const reversedChars = [];
	for (let i = string.length - 1; i >= 0; i--){
		reversedChars.push(string[i]);
	}
	return string === reversedChars.join("");
}

// O(n) Time and Space
function isPalindrome(string, i = 0) {
	const j = string.length - 1 - i;
	return i >= j ? true : string[i] === string[j] && isPalindrome(string, i + 1);
}



// O(n) Time and O(1) Space
function isPalindrome(string) {
	let leftIdx = 0;
	let rightIdx = string.length - 1;
	while(leftIdx < rightIdx){
		if(string[leftIdx] !== string[rightIdx]) return false;
		leftIdx++;
		rightIdx--;
	}
	return true;
}

File: 013 Palindome Check/Palindrome check.py
# O(n) time | O(n) space
def isPalindrome(string):
    reversedChars = []
	for x in reversed(range(len(string))):
		reversedChars.append(string[x])
	return string == "".join(reversedChars)

# O(n) time | O(n) space
def isPalindrome(string, i=0):
	j = len(string) - 1 - i
	return True if i >= j else string[i] == string[j] and isPalindrome(string, i+1)

# O(n) time | O(1) space
def isPalindrome(string):
	left_index = 0 
	right_index = len(string)-1
	while left_index < right_index:
		if string[left_index] != string[right_index]:
			return False
		left_index += 1
		right_index -= 1
	return True